### ISSUE TITLE
Linters not ignoring wasm files with go 1.11 and golangci-lint 1.10

### ISSUE THREAD

**Issue by alecthomas** [golangci/golangci-lint#206]
It looks to me like `_wasm.go` isn't being detected as an architecture suffix and is thus being compiled along with `_amd64.go`, resulting in duplicate symbols. Note that this doesn't occur with golangci-lint 1.9.

---

Thank you for creating the issue!

Please include the following information:

1. Version of golangci-lint: `golangci-lint --version` (or git commit if you don't use binary distribution)

golangci-lint has version 1.10 built from a2b9012 on 2018-08-12T19:35:03Z

2. Go environment: `go version && go env`

```
go version go1.11 darwin/amd64
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/aat/Library/Caches/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/aat/.go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/Cellar/go/1.11/libexec"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.11/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/db/mz_hvw_x54q6mcrbgmslnq7400042q/T/go-build810566632=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

3. Verbose output of running: `golangci-lint run -v`

```
golangci-lint run -v ./...
INFO [config_reader] Config search paths: [./ /Users/aat/Projects/kong /Users/aat/Projects /Users/aat /Users /]
INFO [config_reader] Used config file .golangci.yml
INFO [config_reader] Config search paths: [./ /Users/aat/Projects/kong /Users/aat/Projects /Users/aat /Users /]
INFO [config_reader] Used config file .golangci.yml
INFO [lintersdb] Active 22 linters: [deadcode depguard dupl errcheck gas goconst gocyclo gofmt goimports golint govet ineffassign interfacer megacheck misspell nakedret prealloc structcheck typecheck unconvert unparam varcheck]
INFO [load/path_resolver] Skipped dirs: [.circleci .git .idea _examples]
INFO [load/path_resolver] Paths resolving took 1.932988ms: [. .]
INFO [load] Program loading took 2.283201159s
INFO [load] SSA repr building took 85.383217ms
INFO [load] Not compiling packages: [internal/cpu runtime runtime/internal/sys runtime/internal/atomic]
WARN [runner/megacheck] Can't run megacheck because of compilation errors in packages [internal/cpu runtime runtime/internal/sys runtime/internal/atomic]: ../../../../../../../usr/local/Cellar/go/1.11/libexec/src/internal/cpu/cpu_x86.go:9: CacheLineSize redeclared in this block and 67 more errors: run `golangci-lint run --no-config --disable-all -E typecheck` to see all errors
INFO [runner] worker.6 took 43.761287ms with stages: misspell: 42.533565ms, nakedret: 1.163912ms, depguard: 15.952µs
INFO [runner] worker.4 took 44.166287ms with stages: govet: 34.045782ms, varcheck: 5.199756ms, deadcode: 2.956882ms, goconst: 1.09802ms
INFO [load/astcache] Parse AST for file ../../../../../../../usr/local/Cellar/go/1.11/libexec/src/internal/cpu/cpu_x86.go on demand
INFO [load/astcache] Parse AST for file ../../../../../../../usr/local/Cellar/go/1.11/libexec/src/internal/cpu/cpu_wasm.go on demand
INFO [load/astcache] Parse AST for file ../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/sys_x86.go on demand
INFO [load/astcache] Parse AST for file ../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/sys_wasm.go on demand
INFO [load/astcache] Parse AST for file ../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/sys/arch_wasm.go on demand
INFO [load/astcache] Parse AST for file ../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/sys/arch_amd64.go on demand
INFO [load/astcache] Parse AST for file ../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/atomic/atomic_wasm.go on demand
INFO [load/astcache] Parse AST for file ../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/atomic/atomic_amd64x.go on demand
INFO [load/astcache] Parse AST for file ../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/atomic/stubs.go on demand
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/internal/cpu/cpu_x86.go:9:7: CacheLineSize redeclared in this block (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/internal/cpu/cpu_wasm.go:7:7: other declaration of CacheLineSize (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/sys_x86.go:16:6: gostartcall redeclared in this block (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/sys_wasm.go:31:6: other declaration of gostartcall (typecheck)
INFO [runner] worker.5 took 51.414348ms with stages: unparam: 19.576904ms, ineffassign: 16.569448ms, structcheck: 8.801398ms, unconvert: 2.011692ms, errcheck: 1.541553ms, gocyclo: 1.220469ms, prealloc: 1.142901ms, typecheck: 292.409µs
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/sys/arch_wasm.go:8:2: ArchFamily redeclared in this block (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/sys/arch_amd64.go:8:2: other declaration of ArchFamily (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/sys/arch_wasm.go:9:2: BigEndian redeclared in this block (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/sys/arch_amd64.go:9:2: other declaration of BigEndian (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/sys/arch_wasm.go:10:2: CacheLineSize redeclared in this block (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/sys/arch_amd64.go:10:2: other declaration of CacheLineSize (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/atomic/atomic_wasm.go:14:6: Load redeclared in this block (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/atomic/atomic_amd64x.go:13:6: other declaration of Load (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/atomic/atomic_wasm.go:20:6: Loadp redeclared in this block (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/atomic/atomic_amd64x.go:19:6: other declaration of Loadp (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/atomic/atomic_wasm.go:26:6: Load64 redeclared in this block (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/atomic/atomic_amd64x.go:25:6: other declaration of Load64 (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/atomic/stubs.go:12:6: Cas redeclared in this block (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/atomic/stubs.go:15:6: Casp1 redeclared in this block (typecheck)
../../../../../../../usr/local/Cellar/go/1.11/libexec/src/runtime/internal/atomic/stubs.go:18:6: Casuintptr redeclared in this block (typecheck)
camelcase.go:89:2: naked return in func `camelCase` with 43 lines of code (nakedret)
INFO [runner] worker.1 took 57.797612ms with stages: gofmt: 52.007659ms, megacheck: 335.889µs
INFO [runner] worker.2 took 65.913435ms with stages: dupl: 65.778939ms
INFO [runner] worker.8 took 66.915858ms with stages: gas: 49.081494ms, interfacer: 17.801152ms
INFO [runner] worker.3 took 99.44018ms with stages: golint: 99.361296ms
INFO [runner] worker.7 took 261.301794ms with stages: goimports: 261.263249ms
INFO [runner] Workers idle times: #1: 203.569747ms, #2: 195.406144ms, #3: 161.894853ms, #4: 217.163239ms, #5: 208.868977ms, #6: 217.547071ms, #8: 194.414803ms
INFO [runner] processing took 18.078748ms with stages: nolint: 13.201233ms, autogenerated_exclude: 2.867882ms, exclude: 1.034076ms, path_prettifier: 872.065µs, uniq_by_line: 31.956µs, max_per_file_from_linter: 22.635µs, max_same_issues: 21.487µs, cgo: 17.29µs, max_from_linter: 6.891µs, diff: 1.868µs, skip_files: 1.365µs
INFO [text_printer] Found 20 issues
INFO [text_printer] Extracting issued lines took 0s
INFO Memory: 27 samples, avg is 214.9MB, max is 366.4MB
INFO Execution took 2.661361704s
```

**Comment by alecthomas** [comment]
I tried `golangci-lint` master and I didn't see this issue!

**Comment by Hawken94** [comment]
@alecthomas But gometalinter has the alike issue!

**Comment by jirfag** [comment]
hi! thank you for reporting,
now golangci-lint built by go1.11: it will fix _wasm.go problem
